driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java

```diff
@@ -114,10 +120,12 @@ public <T> Mono<T> execute(final AsyncWriteOperation<T> operation, final ReadCon
 
         return Mono.from(subscriber ->
                 clientSessionHelper.withClientSession(session, this)
-                        .map(clientSession -> getReadWriteBinding(getContext(subscriber), primary(), readConcern,
-                                clientSession, session == null && clientSession != null))
+                        .map(clientSession -> getReadWriteBinding(getContext(subscriber),
+                                primary(), readConcern, clientSession, session == null))
+                        // TODO (CSOT) - does this get called - can the above return empty?
```

Comment:
As `clientSessionHelper.withClientSession` should always returns a session - either the users session or a generated one, the `switchIfEmpty` isn't needed - added #1455 to confirm.

Reply:
Thank you for the clarification and for addressing the TODOs. I approved the https://github.com/mongodb/mongo-java-driver/pull/1455.